pkg/daemons/agent/agent.go

```diff
@@ -194,12 +194,27 @@ func checkCgroups() (root string, hasCFS bool, hasPIDs bool) {
 				if _, err := os.Stat(p); err == nil {
 					hasCFS = true
 				}
-			} else if system == "name=systemd" {
+			}
+		}
+	}
+
+	g, err := os.Open("/proc/1/cgroup")
+	if err != nil {
+		return "", false, false
+	}
+	defer g.Close()
+	root = ""
+	scan = bufio.NewScanner(g)
+	for scan.Scan() {
+		parts := strings.Split(scan.Text(), ":")
+		if len(parts) < 3 {
```

Comment:
What does this mean? Unexpected formatting? If so, shouldn't log at warning about it? Some other condition we can safely ignore? If so, please explain or link to a control groups doc that explains.

Reply:
+1 - some inline comments explaining what is being scanned for and why would be helpful.

Reply:
This logic was taken from above; I am assuming that in the event of unexpected formatting our check falls through as the system is not formatting the contents of these `proc` files properly